lib/execution/execution-triple.ts

```diff
@@ -0,0 +1,64 @@
+import os from 'os';
+
+import {BaseExecutionTriple, ExecutionSpecialty} from '../../types/execution/execution-triple.js';
+import {InstructionSet} from '../../types/instructionsets.js';
+
+let _host_specialty = ExecutionSpecialty.cpu;
+
+export function setHostSpecialty(value: ExecutionSpecialty) {
+    _host_specialty = value;
+}
+
+export class ExecutionTriple extends BaseExecutionTriple {
+    constructor() {
+        super();
+        this.initHostTriple();
+    }
+
+    private initHostTriple() {
+        const hostArch = os.arch();
+        if (hostArch === 'x64') {
+            this.instructionSet = 'amd64';
+        } else if (hostArch === 'arm64') {
+            this.instructionSet = 'aarch64';
+        } else if (hostArch === 'arm') {
+            this.instructionSet = 'arm32';
+        } else {
+            this.instructionSet = os.arch() as InstructionSet;
+        }
+
+        this.os = os.platform();
+        this.specialty = _host_specialty;
+    }
+
+    matchesCurrentHost(): boolean {
```

Comment:
I'm a bit confused here... it seems like this would always match? the constructor sets `this.instructionSet` to be the host one?

Presumably it's updated elsewhere? That seems confusing to me? I would hve imagined that there would be a one global `const hostTriple = new ExecutionTriple(os.host(), os.arch(), ... etc)` and then we would "just" do something like:

`hostTriple.matches(requestTriple)` instead of a `matchesCurrentHost()`

Then we can write tests for this without every call to `matchesCurrentHost()` calling `os.blah()` functions which are hard to mock.

Reply:
By default construction it should match yes. But the setters are used when binaries are prepared for execution, and so they will differ based on compilation result.

Reply:
added unittests for some situations

Reply:
(though you still have valid points, might still look into further refactoring)